Improvement of welcoming documentation

aurelien-reeves opened the pull request:
Improvement of README and CONTRIBUTING documents as part of https://github.com/cucumber/common/issues/1521

olleolleolle commented on README.md:
```diff
@@ -33,9 +50,100 @@ See [CONTRIBUTING.md](CONTRIBUTING.md) for info on contributing to Cucumber.
+- The main website: https://cucumber.io/
+- Documentation: https://cucumber.io/docs
+- Ruby API Documentation: http://www.rubydoc.info/github/cucumber/cucumber-ruby/
+- Support forum: https://groups.google.com/group/cukes
```
https://community.smartbear.com/t5/Cucumber-Open/bd-p/CucumberOS is a newer forum, and we should promote that.

aurelien-reeves replied:
Thank you

Yes, there is still room for improvement.
I have not reworked that last part indeed

aurelien-reeves commented on README.md:
````diff
@@ -33,9 +50,100 @@ See [CONTRIBUTING.md](CONTRIBUTING.md) for info on contributing to Cucumber.
+And a file named `steps.rb` in `features/step_definitions` with:
+
+```ruby
+Given("this will pass") do
````
Todo: Add `# features/step_definitions/steps.rb` here

olleolleolle commented on CONTRIBUTING.md:
```diff
@@ -20,13 +20,14 @@ following first, in order to get your bearings.
 
 You can also join the Cucumber Community Slack team:
 [register for an account][register-slack] then head over to [#intro][slack-intro].
+This is the place to be to meet the mainteners aka the core team.
```
```suggestion
This is the place to be to meet the maintainers aka the core team.
```

aurelien-reeves replied:
Thanks @olleolleolle 🙂

I'll fix that 👍

olleolleolle replied:
I _really_ like what you are doing here. These texts are doing a big job.

aurelien-reeves replied:
Thank you ☺️

mattwynne commented:
This looks great! I wonder if we should explain more about the relationship with the core gem, and how to set up a development environment that includes both.

mattwynne commented:
There's also nothing mentioned (unless I missed it) about the standards we expect from PR submissions - TDD/test coverage, rubocop linting etc. That might be helpful. We could also explicitly point out that we full-time maintainers are available for 1:1 pairing sessions to coach people with their submissions. I'd happily make my calendly link available in this document.

mattwynne commented:
Maybe that latter stuff belongs more in a [PR template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates)

mattwynne commented:
Relevant: https://docs.github.com/en/communities

aurelien-reeves commented:
> This looks great! I wonder if we should explain more about the relationship with the core gem, and how to set up a development environment that includes both.

I though that it would be more relevant in the `core` repo, or maybe in an ARCHITECTURE.md shared across the cucumber-ruby repositories. Or maybe in the wiki?


> There's also nothing mentioned (unless I missed it) about the standards we expect from PR submissions - TDD/test coverage, rubocop linting etc. That might be helpful. 

Actually I wanted to put that in the PR template. And remain pretty gentle with this. There are not too many PR, we can guide contributors through "our" expected standards as part of the reviews. That would let the welcoming document neater.

> We could also explicitly point out that we full-time maintainers are available for 1:1 pairing sessions to coach people with their submissions. I'd happily make my calendly link available in this document.

Yes, that could be interesting indeed 👍 


To sum-up, my strategy here is to to avoid the README and the CONTRIBUTING to be overwhelming but to distill the info all along the process a new contributor would follow when sending a first PR.

mattwynne commented:
Yes, I like the layered approach, making this entry-point doc not too long and leaving depth to other docs. From having paired with @eduardrudko on [his PR](https://github.com/cucumber/cucumber-ruby-core/pull/218), I could see that the two-gem situation caused him some confusion at first, and the setup with the environment variables needs to be documented. Maybe instead of documenting it we should just bite the bullet and merge the two repos into a one little cucumber-ruby monorepo with both gems in it?

mattwynne commented:
> Actually I wanted to put that in the PR template. And remain pretty gentle with this.

Oh yes, I agree we should be gentle and constructive. I think that setting expectations can be kind and supportive - because it will be frustrating for someone to make a PR and then be told later "could you write some tests for this please?"

aurelien-reeves commented:
> the core gem, and how to set up a development environment that includes both.

Just added a new chapter dedicated to cucumber-ruby-core. For now it do not say that much. I suggest to complete it once we have more document related to it in cucumber-ruby-core, or in the wiki, or somewhere else. I would avoid to put too many info here.

aurelien-reeves commented:
Refs cucumber/docs/pull/635

> 
> 
> There's also nothing mentioned (unless I missed it) about the standards we expect from PR submissions - TDD/test coverage, rubocop linting etc. That might be helpful. We could also explicitly point out that we full-time maintainers are available for 1:1 pairing sessions to coach people with their submissions. I'd happily make my calendly link available in this document.

I've updated the PR template, and added some refs. to full-time maintainers.

mattwynne commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,9 +1,40 @@
+
+- [ ] Tests to check the changes in that PR have been added
+- [ ] New and existing tests are passing locally and on the CI
+- [ ] `bundle exec rubocop` reports no offense
```
Does rubocop not run as part of CI?

aurelien-reeves replied:
Yes it is.
But it is also really frustrating to have the CI failing because of rubocop.

This is a way to warn the developer before it happened.

mattwynne reviewed (approved):
My overall thoughts about this change are:

1. It makes things a lot better, and we should merge it ASAP
2. The contributing file is still very long, and I wonder if we can push some of the details off into other files in the /docs folder to keep it more concise and therefore approachable.
3. The example in the README (features/rule.feature) is a bit contrived / self-referential and might not do a great job of illustrating how to use Cucumber. I wonder if we should just link to the https://cucumber.io/docs/guides/10-minute-tutorial/ instead

In any case, I'd merge this and we can iterate from there.

I'd really love to do some user testing and get feedback from people.

aurelien-reeves commented:
>     2. The contributing file is still very long, and I wonder if we can push some of the details off into other files in the /docs folder to keep it more concise and therefore approachable.
> 

Yeah, I agree.
I still like the approach of nextcloud. Their main "how to contribute" page is hosted on their website: https://nextcloud.com/contribute/
That would free the CONTRIBUTING.md in the repo from anything else than contribution dedicated to the code.

>     3. The example in the README (features/rule.feature) is a bit contrived / self-referential and might not do a great job of illustrating how to use Cucumber. I wonder if we should just link to the https://cucumber.io/docs/guides/10-minute-tutorial/ instead

The purpose of that example in the readme is to show a minimal example to show that cucumber is properly installed and running. The minimal reproducible example of it being working 😅

It is not an illustration of how to use Cucumber.

For illustrating how to use Cucumber, the section after that is redirecting to the documentation itself.

> I'd really love to do some user testing and get feedback from real users of this stuff.

+1
